PayloadPkg/Library/PayloadSupportLib/PayloadSupportLib.c

```diff
@@ -90,8 +90,9 @@ ParseAcpiTableInfo (
   EFI_STATUS                                    Status;
 
   Rsdp = (EFI_ACPI_3_0_ROOT_SYSTEM_DESCRIPTION_POINTER *) (UINTN)AcpiBase;
+  ASSERT (Rsdp);  
   Xsdt = (EFI_ACPI_DESCRIPTION_HEADER *) (UINTN) (Rsdp->XsdtAddress);
-  ASSERT (Rsdp || Xsdt);
+  ASSERT (Xsdt);
```

Comment:
In general,  let us try to use less ASSERT() if possible.  Each ASSERT will cause extra size to the final image.
In this case,  I think better to 

Rsdp = (EFI_ACPI_3_0_ROOT_SYSTEM_DESCRIPTION_POINTER *) (UINTN)AcpiBase;
if (Rsdp != NULL) {
   Xsdt = (EFI_ACPI_DESCRIPTION_HEADER *) (UINTN) (Rsdp->XsdtAddress);
} else {
  Xsdt = NULL;
}
ASSERT (Xsdt != NULL);